homeassistant/components/orangepi_gpio/binary_sensor.py

```diff
@@ -61,6 +68,6 @@ def is_on(self):
         """Return the state of the entity."""
         return self._state != self._invert_logic
 
-    def update(self):
-        """Update the GPIO state."""
-        self._state = orangepi_gpio.read_input(self._port)
+    async def async_update(self):
+        """Update state with new GPIO data."""
+        self._state = read_input(self._port)
```

Comment:
This does blocking I/O.

We're not allowed to do sync I/O in coroutines, in the event loop.

Please keep the update method a regular function.